packages/admin/cms-admin/src/dependencies/createDocumentDependencyMethods.ts

```diff
@@ -0,0 +1,98 @@
+import { gql } from "@apollo/client";
+import { BlockInputApi, BlockInterface } from "@comet/blocks-admin";
+
+import { GQLPageTreeNode, Maybe } from "../graphql.generated";
+import { DependencyInterface } from "./types";
+
+interface Query<RootBlocks extends Record<string, BlockInterface>> {
+    node: Maybe<
+        { id: string; pageTreeNode: Maybe<Pick<GQLPageTreeNode, "id" | "category">> } & { [Key in keyof RootBlocks]: BlockInputApi<RootBlocks[Key]> }
+    >;
+}
+
+interface QueryVariables {
+    id: string;
+}
+
+export function createDocumentDependencyMethods<RootBlocks extends Record<string, BlockInterface>>({
+    rootQueryName,
+    rootBlocks,
+    basePath,
+}: {
+    rootQueryName: string;
+    rootBlocks: { [Key in keyof RootBlocks]: RootBlocks[Key] | { block: RootBlocks[Key]; path?: string } };
+    basePath:
+        | string
+        | ((
+              node: { id: string; pageTreeNode: Pick<GQLPageTreeNode, "id" | "category"> } & {
+                  [Key in keyof RootBlocks]: BlockInputApi<RootBlocks[Key]>;
+              },
+          ) => string);
+}): Pick<DependencyInterface, "resolveUrl"> {
+    return {
+        resolveUrl: async ({ rootColumnName, jsonPath, apolloClient, id }) => {
+            const { data, error } = await apolloClient.query<Query<RootBlocks>, QueryVariables>({
+                query: gql`
+                    query ${rootQueryName}Dependency($id: ID!) {
+                        node: ${rootQueryName}(id: $id) {
+                            id
+                            ${Object.keys(rootBlocks).join("\n")}
```

Comment:
That's nice because it removes the need to manually define the query. I didn't realise this was possible